File: LogisimPlus/LogisimPlus/logisim/gui/prefs/PrefBoolean.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.prefs
{

	using logisim.prefs;
	using StringGetter = logisim.util.StringGetter;

	internal class PrefBoolean : JCheckBox, ActionListener, PropertyChangeListener
	{
		private PrefMonitor<bool> pref;
		private StringGetter title;

		internal PrefBoolean(PrefMonitor<bool> pref, StringGetter title) : base(title.get())
		{
			this.pref = pref;
			this.title = title;

			addActionListener(this);
			pref.addPropertyChangeListener(this);
			setSelected(pref.Boolean);
		}

		internal virtual void localeChanged()
		{
			setText(title.get());
		}

		public virtual void actionPerformed(ActionEvent e)
		{
			pref.Boolean = this.isSelected();
		}

		public virtual void propertyChange(PropertyChangeEvent @event)
		{
			if (pref.isSource(@event))
			{
				setSelected(pref.Boolean);
			}
		}
	}

}

File: logisim/app/src/main/java/logisim/gui/prefs/PrefBoolean.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.prefs;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.beans.PropertyChangeEvent;
import java.beans.PropertyChangeListener;

import javax.swing.JCheckBox;

import logisim.prefs.PrefMonitor;
import logisim.util.StringGetter;

class PrefBoolean extends JCheckBox implements ActionListener, PropertyChangeListener {
	private PrefMonitor<Boolean> pref;
	private StringGetter title;

	PrefBoolean(PrefMonitor<Boolean> pref, StringGetter title) {
		super(title.get());
		this.pref = pref;
		this.title = title;

		addActionListener(this);
		pref.addPropertyChangeListener(this);
		setSelected(pref.getBoolean());
	}

	void localeChanged() {
		setText(title.get());
	}

	public void actionPerformed(ActionEvent e) {
		pref.setBoolean(isSelected());
	}

	public void propertyChange(PropertyChangeEvent event) {
		if (pref.isSource(event)) setSelected(pref.getBoolean());
	}
}
